### PR TITLE
`@ponder/client` db

### PR DESCRIPTION
**Pull request by kyscott18**
## `@ponder/client`
```ts
import { createClient } from "@ponder/client";
import * as schema from "../../ponder/ponder.schema.ts";

const client = createClient("http://localhost:42069", { schema });

const result = await client.db.select().from(schema.account).limit(10);
```

### CODE REVIEW

File: docs/pages/docs/query/client.mdx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+### Register client middleware
+
+```ts filename="src/api/index.ts" {7}
+import { db } from "ponder:api";
+import { Hono } from "hono";
+import { client } from "ponder";
+
+const app = new Hono();
+
+app.use(client({ db }));
```
Comment (typedarray) [inline docs/pages/docs/query/client.mdx:36]: Does it make sense to expose to the user which path the endpoint gets mounted at?

File: docs/pages/docs/query/client.mdx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+## Get started
+
+<Steps>
+
+### Update to `>=0.9.0`
+
+`@ponder/client` is available starting from version `0.9.0`. Read the [migration guide](/docs/migration-guide#090) for more details.
+
+### Register client middleware
+
```
Comment (typedarray) [inline docs/pages/docs/query/client.mdx:28]: Register the `client` middleware to enable client queries.

File: docs/pages/docs/query/client.mdx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```ts filename="src/api/index.ts" {7}
+import { db } from "ponder:api";
+import { Hono } from "hono";
+import { client } from "ponder";
```
Comment (typedarray) [inline docs/pages/docs/query/client.mdx:32]: I wonder if this and `graphql` should come from a `ponder/middleware` entrypoint

File: docs/pages/docs/query/client.mdx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+const app = new Hono();
+
+app.use(client({ db }));
+
+export default app;
+```
+
+### Install `@ponder/client`
+
```
Comment (typedarray) [inline docs/pages/docs/query/client.mdx:42]: Install the `@ponder/client` package in your client project. This package works in any JavaScript environment, including the browser, server-side scripts, and both client and server code from web frameworks like Next.js

File: docs/pages/docs/query/client.mdx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+const client = createClient("https://...", { schema });
+
+const result = await client.db.select().from(schema.account).limit(10);
+```
+
+</Steps>
+
+## API reference
+
+### `client.db`
```
Comment (typedarray) [inline docs/pages/docs/query/client.mdx:77]: Would be cool to fit `status` in here somehow. API looks good for now though.

File: docs/pages/docs/query/client.mdx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+  </Tabs.Tab>
+</Tabs>
+
+### Create a client and send a request
+
+```ts {4,6}
+import { createClient } from "@ponder/client";
+import * as schema from "../../ponder/ponder.schema";
+
+const client = createClient("https://...", { schema });
```
Comment (typedarray) [inline docs/pages/docs/query/client.mdx:68]: Should make it more clear what HTTP URL you need to pass here - root path of the ponder server?

File: docs/pages/docs/query/direct-sql.mdx [hunk 7]
```diff
@@ -113,18 +113,19 @@ SELECT "to", COUNT(*) AS transfer_count
...
-Here's a script that creates a Drizzle client and runs a query against the Ponder tables. Be sure to connect to the database using the same `DATABASE_URL` as the Ponder app.
+Here's a script that creates a Drizzle client and runs a query against the Ponder tables in the specified schema. Be sure to connect to the database using the same `DATABASE_URL` as the Ponder app.
 
 ```ts filename="query.ts"
+import { setDrizzleSchema } from "@ponder/client";
 import { drizzle } from "drizzle-orm/node-postgres";
-import * as schema from "../../ponder/ponder.schema";
+import * as _schema from "../../ponder/ponder.schema";
+
+const schema = setDrizzleSchema(_schema, "my_schema");
```
Comment (typedarray) [inline docs/pages/docs/query/direct-sql.mdx:128]: `setDatabaseSchema` ? I tried to use the term "database schema" all over the docs in the last release. The JSDoc could explain what the function actually does "update Drizzle table and enum objects to use the specific database schema / namespace"

File: examples/with-nextjs/frontend/src/hooks/useDeposits.ts [hunk 8]
```diff
@@ -1,47 +1,24 @@
...
-        }[];
-      };
-      return r.depositEvents.map((d) => ({
-        id: d.id,
-        timestamp: d.timestamp,
-        account: d.account,
-        amount: BigInt(d.amount),
-      }));
-    },
+    queryFn: async () =>
```
Comment (typedarray) [inline examples/with-nextjs/frontend/src/hooks/useDeposits.ts:12]: rm `async` ?

File: examples/with-nextjs/ponder/ponder.schema.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { onchainTable } from "ponder";
+import { onchainTable } from "ponder/schema";
```
Comment (typedarray) [inline examples/with-nextjs/ponder/ponder.schema.ts:1]: Honestly I could see us removing `onchainTable` from the root entrypoint so users get the tree-shaking benefit for free. Tiny breaking change in the grand scheme of things.

Do worry a bit about `ponder/schema` vs `ponder:schema` though ...

File: docs/pages/docs/query/client.mdx (hunk 1 above)
Comment (kyscott18) [reply docs/pages/docs/query/client.mdx:36]: I don't think it does, just makes it easier to mess up

**Pull request merged**